File: CPP/Program/Clothing.cpp
#include <iostream>
#include <string>
#include "Product.cpp"

using namespace std;

// Deklarasi kelas Clothing yang merupakan turunan dari kelas Product
class Clothing : public Product {
private:
    string size;
    string material;
    string gender;

    // Atribut statis untuk menyimpan panjang maksimum dari setiap atribut
    static int maxSize;
    static int maxMaterial;
    static int maxGender;

public:
    // Konstruktor default Clothing yang memanggil konstruktor default dari kelas Product
    Clothing() : Product() {
        size = "";
        material = "";
        gender = "";
    }

    // Konstruktor Clothing dengan parameter yang memanggil konstruktor parameter dari kelas Product
    Clothing(string id, string name, string brand, string price, string size, string material, string gender)
        : Product(id, name, brand, price), size(size), material(material), gender(gender) {

        // Memperbarui nilai maksimum jika panjang atribut melebihi nilai maksimum sebelumnya
        if (this->size.length() > maxSize) {
            maxSize = this->size.length();
        }

        if (this->material.length() > maxMaterial) {
            maxMaterial = this->material.length();
        }

        if (this->gender.length() > maxGender) {
            maxGender = this->gender.length();
        }
    }

    // Getter dan Setter untuk atribut size
    string get_size() {
        return size;
    }

    void set_size(string size) {
        this->size = size;
    }

    // Getter dan Setter untuk atribut material
    string get_material() {
        return material;
    }

    void set_material(string material) {
        this->material = material;
    }

    // Getter dan Setter untuk atribut gender
    string get_gender() {
        return gender;
    }

    void set_gender(string gender) {
        this->gender = gender;
    }

    // Getter untuk nilai maksimum panjang atribut size
    static int getmaxSize() {
        return maxSize;
    }

    // Getter untuk nilai maksimum panjang atribut material
    static int getmaxMaterial() {
        return maxMaterial;
    }

    // Getter untuk nilai maksimum panjang atribut gender
    static int getmaxGender() {
        return maxGender;
    }
};

File: Python/Program/Clothing.py
# Clothing.py

from Product import Product

class Clothing(Product):
    # Variabel statis untuk meyimpan panjang maksimum setiap atribut Clothing
    max_size = 0
    max_material = 0
    max_gender = 0

    def __init__(self, id, name, brand, price, size, material, gender):
        """
        Menginisialisasi objek Clothing dengan atribut yang diberikan.

        Parameter:
        - id: Identifikasi unik untuk produk pakaian.
        - name: Nama produk pakaian.
        - brand: Merek produk pakaian.
        - price: Harga produk pakaian.
        - size: Ukuran produk pakaian.
        - material: Bahan produk pakaian.
        - gender: Jenis kelamin produk pakaian.
        """
        super().__init__(id, name, brand, price)
        self.size = size
        self.material = material
        self.gender = gender

        # Memperbarui panjang maksimum jika panjang atribut saat ini lebih besar
        if len(size) > Clothing.max_size:
            Clothing.max_size = len(size)

        if len(material) > Clothing.max_material:
            Clothing.max_material = len(material)

        if len(gender) > Clothing.max_gender:
            Clothing.max_gender = len(gender)

    # Metode Getter dan Setter untuk setiap atribut Clothing
    def get_size(self):
        return self.size

    def set_size(self, size):
        self.size = size

    def get_material(self):
        return self.material

    def set_material(self, material):
        self.material = material

    def get_gender(self):
        return self.gender

    def set_gender(self, gender):
        self.gender = gender

    # Metode Getter untuk panjang maksimum setiap atribut Clothing
    @staticmethod
    def get_max_size():
        return Clothing.max_size

    @staticmethod
    def get_max_material():
        return Clothing.max_material

    @staticmethod
    def get_max_gender():
        return Clothing.max_gender
